ops/pebble.py

```diff
@@ -308,29 +317,97 @@ def __repr__(self):
                 ).format(self=self)
 
 
-class API:
+class Setup:
+    """Represents a Pebble setup layer (or flattened setup).
+
+    This is not documented, but captured in code here:
+    https://github.com/canonical/pebble/blob/master/internal/setup/setup.go
+    """
+
+    def __init__(self, raw: Union[str, Dict] = None):
+        if isinstance(raw, str):
+            d = yaml.safe_load(raw) or {}
+        else:
+            d = raw or {}
+        self.summary = d.get('summary', '')
+        self.description = d.get('description', '')
+        self.services = {name: SetupService(name, service)
+                         for name, service in d.get('services', {}).items()}
+
+    def to_yaml(self) -> str:
+        """Convert this setup to its YAML representation."""
+        return yaml.dump(self.to_dict(), Dumper=_safe_dumper)
+
+    def to_dict(self) -> Dict:
+        """Convert this setup to a dict representation."""
+        fields = [
+            ('summary', self.summary),
+            ('description', self.description),
+            ('services', {name: service.to_dict() for name, service in self.services.items()})
+        ]
+        return {name: value for name, value in fields if value}
+
+    def __repr__(self) -> str:
+        return 'Setup({!r})'.format(self.to_dict())
+
+    __str__ = to_yaml
+
+
+class SetupService:
```

Comment:
That would sound better as PebbleService, I think, or pebble.Service, if we're buying into that convention (pebble.Setup or pebble.Layer probably means we're already there).

Reply:
Changed to `pebble.Service`.